methods.rb

```diff
@@ -0,0 +1,116 @@
+module Enumerable
+  def my_each
+    i = 0
+    length.times do
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_map(proc = nil)
+    temp = []
+    i = 0
+    if proc.nil?
+      length.times do
+        temp.push(yield(self[i]))
+        i += 1
+      end
+    else
+      length.times do
+        temp.push(proc.call(self[i]))
+        i += 1
+      end
+    end
+    temp
+  end
+
+  def my_each_with_index
+    i = 0
+    length.times do
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    temp = []
+    i = 0
+    length.times do
+      temp.push(self[i]) if yield(self[i])
+      i += 1
+    end
+    temp
+  end
```

Comment:
- [ ] `my_select` is not working as the original `select` method, also is not passing all test cases.

### `select` is an Enumerable that has these characteristics:
- Returns an Enumerator if no block is given.
```
[1, 2, 3].select
# #<Enumerator: [1, 2, 3]:select>
```
- Returns a **new array** with the results that match the condition in the block
```
[1, 2, 3].select { |num| num > 2 }
# [3]
```
- doesn't change/modify the original array/range/hash

[Enumerable#select](https://ruby-doc.org/core-2.7.2/Enumerable.html#method-i-select)